includes/meta.php

```diff
@@ -83,46 +125,97 @@ function register_meta_boxes() {
  * @param array   $meta_box The settings for the meta box, including an 'args' key.
  */
 function render_meta_box( $post, $meta_box ) {
+
+	// Bail if no args are passed. Shouldn't get this far, but ¯\_(ツ)_/¯.
+	if ( empty( $meta_box['args'] ) ) {
+		return;
+	}
+
+	// Fetch any selected terms.
 	$selected = array_map( 'intval', get_post_meta( $post->ID, '_sticky_tax' ) );
+
+	// Set an empty options array.
 	$options  = array();
-	$size     = 0;
-
-	foreach ( $meta_box['args'] as $taxonomy ) {
-		$terms = get_terms( [
-			'taxonomy'   => $taxonomy,
-			'fields'     => 'id=>name',
-			'hide_empty' => false,
-		] );
-
-		if ( ! empty( $terms ) ) {
-			$tax                    = get_taxonomy( $taxonomy );
-			$options[ $tax->label ] = $terms;
-			$size += count( $terms ) + 1;
-		}
+
+	// Loop the taxonomy data provided.
+	foreach ( $meta_box['args'] as $tax_name => $tax_label ) {
+
+		// Fetch my post terms.
+		$terms = wp_get_post_terms( $post->ID, $tax_name, array() );
+
+		// Set our items based on whether we have selected items or not.
+		$items  = empty( $terms ) || is_wp_error( $terms ) ? '' : wp_list_pluck( $terms, 'name', 'term_id' );
+
+		// Set my options array accordingly.
+		$options[ $tax_name ] = [
+			'label' => $tax_label,
+			'items' => $items,
+		];
 	}
 
-	// Limit the <select> size.
-	$size = 10 > $size ? $size : 10;
-?>
+	// Bail without options data to show.
+	if ( empty( $options ) ) {
+		return;
+	}
 
-	<label for="sticky-tax-term-id" class="screen-reader-text"><?php esc_html_e( 'Sticky terms', 'sticky-tax' ); ?></label>
-	<p>
-		<select id="sticky-tax-term-id" name="sticky-tax-term-id[]" class="regular-text" size="<?php echo esc_attr( $size ); ?>" multiple>
-			<?php foreach ( $options as $group => $opts ) : ?>
-				<optgroup label="<?php echo esc_attr( $group ); ?>">
-					<?php foreach ( $opts as $id => $label ) : ?>
+	// Wrap the whole thing in a div.
+	echo '<div class="term-sticky-list-wrap">';
 
-						<option value="<?php echo esc_attr( $id ); ?>" <?php selected( in_array( $id, $selected, true ), true ); ?>><?php echo esc_html( $label ); ?></option>
+	// Loop the grouped term items.
+	foreach ( $options as $type => $data ) {
 
-					<?php endforeach; ?>
-				</optgroup>
-			<?php endforeach; ?>
-		</select>
-	</p>
+		// Set a class for the message.
+		$class  = ! empty( $data['items'] ) ? 'term-sticky-list-hide' : 'term-sticky-list-show';
 
-	<p class="description"><?php esc_html_e( '"Stick" this post to the top of the archive pages for the selected term(s)', 'sticky-tax' ); ?></p>
+		// Wrap each group in a div.
+		echo '<div id="term-sticky-' . esc_attr( $type ) . '-group" class="term-sticky-list-group" data-term-type="' . esc_attr( $type ) . '" >';
```

Comment:
More a matter of personal preference, but I'm curious: how would you feel about reverting this back to a block of HTML embedded within the PHP, rather than a series of echo statements? I've found that in situations of >= 4 lines or so it can become much more readable, and IDEs gain the benefit of syntax highlighting on the markup.

For example:
```php
  // Setup work.
?>

  <h4><?php echo esc_html( $data['label'] ); ?></h4>
  <ul id="list-<?php echo esc_attr( $type ); ?>" class="term-sticky-list term-sticky-list-<?php echo esc_attr( $type ); ?>">
    <?php foreach ( $data['items'] as $term_id => $term_name ) : ?>

      <li id="item-<?php echo esc_attr( $term_id ); ?>" data-term-id="<?php echo esc_attr( $term_id ); ?> class="term-sticky-list-item">
        <!-- ... -->
      </li>

    <?php endforeach; ?>
  </ul>

<?php
  // Resume PHP-intensive stuff.
```

Reply:
it's personal preference. I prefer the better syntax highlighting on the PHP functions that happens when echo'd, and it reads more disjointed to me when individual tags are echo'd inside of markup, but this ain't a hill worth dying on.